tests/e2e/test_service.go

```diff
@@ -108,43 +118,42 @@ func (a *alertStore) GetAll() []string {
 	return n
 }
 
-const segmentSize uint64 = 1 << 20
-
-type Segment [segmentSize]*protocol.SignedSnapshot
-
 type snapStore struct {
-	segments []Segment
-	sync.Mutex
+	data *freecache.Cache
 }
 
 func newSnapStore() *snapStore {
-	return &snapStore{segments: append([]Segment{}, Segment{})}
+	return &snapStore{
+		data: freecache.NewCache(2 << 30),
```

Comment:
Maybe a small comment of this "magic number"? (why it's 2<<30 at least)

Reply:
That's a gigabyte of cache. Its not a magic number. Seems good enough for our testing environment. No other particular reason.